### PR TITLE
Added lots of new features to the backend tagging interface

### PR DESCRIPTION
**Pull request by nmaludy**
Additions:
* Added lots of new back-end API functionality to the `vmwarelib.tagging.VmwareTagging` interface.
* Added new `vsphere.tag_attach_bulk` action to attach a tag to a bulk set of resources.
  Primary usecase is tagging all VMs in a cluster dynamically.
* Added new `vsphere.tag_detach_bulk` action to detach a tag from a bulk set of resources.
  Primary usecase is remoave a tag from all VMs in a cluster dynamically.

### CODE REVIEW

File: actions/tag_attach_bulk.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            replace,
+            tag_name, tag_description,
+            vsphere):
+        self.connect_rest(vsphere)
+
+        # create the category and tags
+        category = self.tagging.category_get_or_create(category_name,
+                                                       category_description,
+                                                       category_cardinality,
+                                                       category_types)
```
Comment (bishopbm1) [inline actions/tag_attach_bulk.py:31]: Thinking we might want a return check here like to did in the detach action https://github.com/StackStorm-Exchange/stackstorm-vsphere/pull/76/files#diff-edb2cc6462154be4b1324b535d3342207a94b39c1665de00be65d00a5bb1ffccR28

**Review by bishopbm1 — COMMENTED** [review]
Looks good just one small thing I saw

File: actions/tag_attach_bulk.py (hunk 1 above)
Comment (nmaludy) [reply actions/tag_attach_bulk.py:31]: This is done in the underlying function, if the category doesn't exist one is created and returned.

**Comment by jschoewe** [comment]
LGTM. I tested all the tagging actions to make sure they still work with the changes to tagging.py and everything ran as expected

**Pull request merged**